### PR TITLE
[SR-5437] Check for untracked files

### PR DESCRIPTION
**Pull request by noefroidevaux**
Fix https://bugs.swift.org/browse/SR-5437 by @ghn and me at SwiftAlps

### CODE REVIEW

File: Sources/SourceControl/GitRepository.swift [hunk 1]
```diff
@@ -302,18 +302,13 @@ public class GitRepository: Repository, WorkingCheckout {
...
-            var nonZeroExit = false
-
-            for args in [args, args + ["--cached"]] {
-                let result = try? Process.popen(arguments: args)
-                nonZeroExit = nonZeroExit || result?.exitStatus != .terminated(code: 0)
-            }
+            let args = [Git.tool, "-C", path.asString, "status", "-s"]
 
-            return nonZeroExit
+            if let result = try? Process.popen(arguments: args), case let .success(value) = result.output {
```
Comment (aciidb0mb3r) [inline Sources/SourceControl/GitRepository.swift]: We have an utility method checkNonZeroExit which might be better here.

File: Sources/SourceControl/GitRepository.swift [hunk 2]
```diff
@@ -302,18 +302,13 @@ public class GitRepository: Repository, WorkingCheckout {
...
-            // This won't detect new untracked files, but it is
-            // just a safety measure for now.
-            let args = [Git.tool, "-C", path.asString, "diff", "--no-ext-diff", "--quiet", "--exit-code"]
-            var nonZeroExit = false
-
-            for args in [args, args + ["--cached"]] {
-                let result = try? Process.popen(arguments: args)
-                nonZeroExit = nonZeroExit || result?.exitStatus != .terminated(code: 0)
-            }
+            let args = [Git.tool, "-C", path.asString, "status", "-s"]
```
Comment (aciidb0mb3r) [inline Sources/SourceControl/GitRepository.swift]: We should add a brief comment explaining what this does.

**Comment by noefroidevaux** [comment]
@aciidb0mb3r We changed the pull request according to your comments

**Comment by aciidb0mb3r** [comment]
@swift-ci please smoke test

**Comment by aciidb0mb3r** [comment]
LGTM, thanks!

**Comment by hartbit** [comment]
@noefroidevaux Thx for sticking to finishing this Pull Request and congrats on the merge!

**Comment by noefroidevaux** [comment]
@hartbit Thanks to you for the workshop at SwiftAlps 😉

**Pull request merged**